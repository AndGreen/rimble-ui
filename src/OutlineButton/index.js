import React from 'react'

import styled from 'styled-components'
import { tint } from 'polished'

import Button from '../Button'

const OutlineButton = styled(Button)`
  & {
  }
  &:hover {
    background-color: ${props => tint(0.9, props.theme.colors.primary)};
  }
`

OutlineButton.defaultProps = {
  color: 'primary',
  bg: 'transparent',
  border: 1,
}

OutlineButton.displayName = 'OutlineButton'

export default OutlineButton
