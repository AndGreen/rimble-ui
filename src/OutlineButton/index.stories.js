import { storiesOf } from '@storybook/react'

import React from 'react'

import OutlineButton from '../OutlineButton'

storiesOf('Button/OutlineButton', module)
  .add('default', () => (
    <div>
      <OutlineButton size='small'>Small OutlineButton</OutlineButton>
      <OutlineButton size='medium'>Medium OutlineButton</OutlineButton>
      <OutlineButton size='large'>Large OutlineButton</OutlineButton>
    </div>
  ))
