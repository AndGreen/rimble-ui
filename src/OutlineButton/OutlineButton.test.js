import OutlineButton from './'

describe('OutlineButton component sanity', () => {
  it('has name', () => {
    expect(OutlineButton.displayName).toBe('OutlineButton')
  })

  it('matches default snapshot', () => {
    const component = render(<OutlineButton />)
    expect(component).toMatchSnapshot()
  })
})
