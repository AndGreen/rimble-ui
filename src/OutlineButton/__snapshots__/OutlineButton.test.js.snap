// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlineButton component sanity matches themed snapshot 1`] = `
.c2 {
  padding-left: 32px;
  padding-right: 32px;
  padding-top: 0px;
  padding-bottom: 0px;
  color: #4E3FCE;
  background-color: #fff;
  height: 48px;
  border: 1px solid;
  border-radius: 4px;
}

.c2 {
  box-sizing: border-box;
  text-align: left;
}

.c1 {
  font-weight: 600;
  font-size: 1rem;
  min-width: 48px;
}

.c1 {
  cursor: pointer;
  font-family: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  line-height: 1;
  overflow: hidden;
  white-space: nowrap;
  -webkit-font-smoothing: antialiased;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all .15s ease;
  transition: all .15s ease;
}

.c1:hover {
  background-color: #5f52d2;
  box-shadow: 0px 2px 4px rgba(0,0,0,0.2);
}

.c1:active {
  background-color: #4638b9;
  box-shadow: none;
}

.c1:disabled {
  opacity: 0.4;
  cursor: not-allowed;
}

.c1 > svg {
  display: block;
}

.c1 > svg:first-child {
  margin-left: -1rem;
  margin-right: .5rem;
}

.c1 > svg:last-child {
  margin-left: .5rem;
  margin-right: -1rem;
}

.c1 > svg:first-child:last-child {
  margin: 0;
}

.c1 {
  background: #FFFFFF;
  border: 1px solid #CCCCCC;
}

.c1:hover {
  color: #7165d7;
  border-color: #7165d7;
  background-color: #FFF;
}

.c1:active {
  color: #3e32a4;
  border-color: #3e32a4;
  background-color: #edebfa;
}

.c0 {
  font-size: 1em;
  font-family: "Source Sans Pro",-apple-system,sans-serif;
  color: #3F3D4B;
}

<div
  class="c0"
>
  <button
    class="c1 c2"
    color="primary"
    font-weight="3"
    height="48px"
    type="button"
  >
    
  </button>
</div>
`;
